nettacker/core/app.py

```
@@ -107,26 +110,36 @@ def expand_targets(self, scan_id):
-                except IndexError:
-                    base_path = ""
-                # remove url proto; uri; port
-                target = target.split("://")[1].split("/")[0].split(":")[0]
-                targets.append(target)
+                    parsed = urlsplit(target)
+                    hostname = parsed.hostname
+
+                    if not hostname:
+                         raise ValueError(
+                               "Invalid scan configuration: target host cannot be empty"
+                         )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)